sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -391,10 +391,11 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   // UDFToBoolean
   private[this] def castToBoolean(from: DataType): Any => Any = from match {
     case StringType =>
+      val dialect = SQLConf.get.getConf(SQLConf.DIALECT)
       buildCast[UTF8String](_, s => {
-        if (StringUtils.isTrueString(s)) {
+        if (StringUtils.isTrueString(s, dialect)) {
```

Comment:
In the case of more dialects, here we can pass the dialect to the `StringUtils` as a parameter to avoid more future changes in Cast.scala.

Reply:
Note that this brings performance overhead.
If anyone objects, we can reconsider create methods for each dialect, or even revert https://github.com/apache/spark/pull/25458